ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/util/DataColumnSidecarELRecoveryManagerImpl.java

```diff
@@ -266,25 +273,31 @@ private void publishRecoveredDataColumnSidecars(
     final int maxCustodyGroups =
         SpecConfigFulu.required(spec.forMilestone(SpecMilestone.FULU).getConfig())
             .getNumberOfCustodyGroups();
-    final List<DataColumnSidecar> myCustodySidecars;
+    final List<DataColumnSidecar> localCustodySidecars;
     if (samplingGroupCount == maxCustodyGroups) {
-      myCustodySidecars = dataColumnSidecars;
+      localCustodySidecars = dataColumnSidecars;
     } else {
-      final Set<UInt64> myCustodyIndices =
+      final Set<UInt64> localCustodyIndices =
           new HashSet<>(custodyGroupCountManagerSupplier.get().getSamplingColumnIndices());
-      myCustodySidecars =
+      localCustodySidecars =
           dataColumnSidecars.stream()
-              .filter(sidecar -> myCustodyIndices.contains(sidecar.getIndex()))
+              .filter(sidecar -> localCustodyIndices.contains(sidecar.getIndex()))
               .toList();
     }
     recoveryTask
         .recoveredColumnIndices()
-        .addAll(myCustodySidecars.stream().map(DataColumnSidecar::getIndex).toList());
+        .addAll(localCustodySidecars.stream().map(DataColumnSidecar::getIndex).toList());
     LOG.debug(
         "Publishing {} data column sidecars for {}",
-        myCustodySidecars::size,
+        localCustodySidecars::size,
         recoveryTask::getSlotAndBlockRoot);
-    dataColumnSidecarPublisher.accept(myCustodySidecars, LOCAL_EL);
+    if (inSync) {
+      dataColumnSidecarPublisher.accept(localCustodySidecars, LOCAL_EL);
```

Comment:
why we dont' send it to subscribers when inSync. If it's intentional, we should focus on this on subscription